### PR TITLE
stylelint

### PR DESCRIPTION
**Pull request by seankoji**
Ported fox sports's existing linting rules from `scss-lint` to stylelint to facilitate usage in [styled-components](https://www.styled-components.com/docs/tooling#stylelint)

### CODE REVIEW

File: rules/limit-language-features.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    'selector-attribute-operator-whitelist': null, // Specify a whitelist of allowed attribute operators.
+    'selector-class-pattern': null, // Specify a pattern for class selectors.
+    'selector-id-pattern': null, // Specify a pattern for id selectors.
+    'selector-max-attribute': 2, // Limit the number of attribute selectors in a selector.
+    'selector-max-class': 3, // Limit the number of classes in a selector.
+    'selector-max-combinators': 2, // Limit the number of combinators in a selector.
+    'selector-max-compound-selectors': 3, // Limit the number of compound selectors in a selector.
+    'selector-max-empty-lines': 0, // Limit the number of adjacent empty lines within selectors.
+    'selector-max-id': 1, // Limit the number of id selectors in a selector.
+    'selector-max-specificity': ["0,4,2", {ignoreSelectors: ['.fsui-normalise', '.fiso-widget']}], // Limit the specificity of selectors.
```
Comment (roubaobaozi) [inline rules/limit-language-features.js:57]: If you have this, max-id, max-class and max-type can be removed :) or the errors will double. And the should be the same, if the separate rules are right, then this should be set to “1,3,2”

File: rules/stylelint-order.js [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+            type: 'rule',
+            selector: /^&/
+        },
+        {
+            type: 'at-rule',
+            name: 'media'
+        },
+        "rules"
+    ],
+    "order/properties-order": [ // Concentric CSS order taken from https://github.com/brandon-rhodes/Concentric-CSS
```
Comment (roubaobaozi) [inline rules/stylelint-order.js:24]: There’s a stylelint concentric order package, do we want to just pull that in rather than maintain our own?

**Review by nirajaryal — APPROVED** [review]

**Review by roubaobaozi — COMMENTED** [review]
Ugh why did my comments not come up? I wrote it ages ago.

File: rules/limit-language-features.js (hunk 1 above)
Comment (roubaobaozi) [reply rules/limit-language-features.js:57]: nevermind, sean’s on holiday, I’m going to make a PR for this :)

Comment (roubaobaozi) [reply rules/limit-language-features.js:57]: … or I can’t, I have no perms :P

Comment (seankoji) [reply rules/limit-language-features.js:57]: you have perms now. I agree it makes more sense to use the rules for each type than this one

File: rules/stylelint-order.js (hunk 2 above)
Comment (seankoji) [reply rules/stylelint-order.js:24]: yes please! I looked around for exactly that before going to all this effort

Comment (roubaobaozi) [reply rules/stylelint-order.js:24]: Cool I’ll make a PR tomorrow :) why are you looking at linting in a Sunday??

Comment (darrennolan) [reply rules/stylelint-order.js:24]: image

I've already yelled at him.... His transition to dazz is now complete.  >_>

Comment (roubaobaozi) [reply rules/stylelint-order.js:24]: 😂

**Pull request merged**